controllers/statusreport/statusreport_adapter.go

```diff
@@ -79,6 +79,16 @@ func (a *Adapter) EnsureSnapshotTestStatusReportedToGitHub() (controller.Operati
 		}
 	}
 
+	if gitops.IsSnapshotMarkedAsPassed(a.snapshot) || gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
```

Comment:
I realized that snapshot can be also marked as invalid

Reply:
Thanks for catching that. Now that I think about it, along with Invalid, the Snapshot's Integration status is also marked as Error but we won't consider it here because the Error state is not a final state and the Snapshot _could_ move out of it. Is that correct?

Reply:
Technically yes, but it may be also forever